### PR TITLE
[ST] Refactor CustomCa STs

### PR DESCRIPTION
**Pull request by jankalinic**
### Type of change

- Refactoring

### Description

In this PR there are a few improvements to Custom Certificate Authority STs.
Additionally, it introduces a newly added / split tests specifically designed to validate manual certificate renewal processes. These tests are designed for verifying reliability of certificate renewals.

### Checklist

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally

### CODE REVIEW

**Comment by jankalinic** [comment]
@strimzi-ci run tests --cluster-type=ocp --cluster-version=4.14 --install-type=bundle --testcase=CustomCaST

File: systemtest/src/test/java/io/strimzi/systemtest/security/custom/CustomCaST.java [hunk 1]
```diff
@@ -627,17 +449,146 @@ void testClientsCaCertificateRenew(ExtensionContext extensionContext) {
...
-                initialKafkaUserCertStartTime.compareTo(changedKafkaUserCertStartTime) < 0);
-        assertThat("UserCert end date has been renewed",
-                initialKafkaUserCertEndTime.compareTo(changedKafkaUserCertEndTime) < 0);
+        // Verify renewal process
+        assertThat("ClientsCA cert should not have changed.", initialCertEndTime.compareTo(changedCertEndTime) == 0);
+        assertThat("UserCert start date has been renewed", initialKafkaUserCertStartTime.compareTo(changedKafkaUserCertStartTime) < 0);
+        assertThat("UserCert end date has been renewed", initialKafkaUserCertEndTime.compareTo(changedKafkaUserCertEndTime) < 0);
+    }
+
+    private void manuallyRenewCa(TestStorage testStorage, SystemTestCertHolder oldCa, SystemTestCertHolder newCa) {
```
Comment (see-quick) [inline systemtest/src/test/java/io/strimzi/systemtest/security/custom/CustomCaST.java:464]: I think this helper method deserves a proper description with steps (Javadoc)

Comment (jankalinic) [reply systemtest/src/test/java/io/strimzi/systemtest/security/custom/CustomCaST.java:477]: Thanks. Added

**Review by im-konge — APPROVED** [review]

**Comment by jankalinic** [comment]
Tests on azp buildId=155378 finished with one error -> testKafkaDoesNotRollsWhenTopicIsUnderReplicated

**Comment by scholzj** [comment]
This is not ready for merge because it does not have the required number of reviews from maintainers.

File: systemtest/src/test/java/io/strimzi/systemtest/security/custom/CustomCaST.java [hunk 2]
```diff
@@ -70,104 +68,70 @@ public class CustomCaST extends AbstractST {
     private static final String STRIMZI_INTERMEDIATE_CA = "C=CZ, L=Prague, O=Strimzi, CN=StrimziIntermediateCA";
 
     @ParallelNamespaceTest
-    void testReplacingCustomClusterKeyPairToInvokeRenewalProcess(ExtensionContext extensionContext) throws IOException, NoSuchAlgorithmException, InvalidKeySpecException {
+    void testReplacingCustomClusterKeyPairToInvokeRenewalProcess(ExtensionContext extensionContext) {
```
[the remark this answers is not in the document]
Comment (jankalinic) [reply systemtest/src/test/java/io/strimzi/systemtest/security/custom/CustomCaST.java:71]: Sure, will add some docs with steps. Thanks

File: systemtest/src/main/java/io/strimzi/systemtest/resources/crd/KafkaResource.java [hunk 3]
```diff
@@ -169,4 +169,12 @@ public static String getKafkaPodName(String clusterName, String nodePoolName, in
...
         return Constants.KAFKA_NODE_POOL_PREFIX + hashStub(clusterName);
     }
+
+    public static void annotateKafka(String clusterName, String namespaceName, Map<String, String> annotations) {
+        replaceKafkaResourceInSpecificNamespace(clusterName, kafka -> kafka.getMetadata().getAnnotations().putAll(annotations), namespaceName);
+    }
+
+    public static void removeAnnotation(String clusterName, String namespaceName, String annotationKey) {
+        replaceKafkaResourceInSpecificNamespace(clusterName, kafka -> kafka.getMetadata().getAnnotations().remove(annotationKey), namespaceName);
+    }
```
[the remark this answers is not in the document]
Comment (jankalinic) [reply systemtest/src/main/java/io/strimzi/systemtest/resources/crd/KafkaResource.java:179]: Considering it should follow other resources, yes, good point, thank you

**Comment by jankalinic** [comment]
@strimzi-ci run tests --cluster-type=minikube --cluster-version=1.27 --install-type=bundle --testcase=CustomCaST

**Review by Frawless — APPROVED** [review]

**Comment by im-konge** [comment]
I accidentally started kraft-regression again, so I cancelled it -> the last successful results are https://dev.azure.com/cncf/strimzi/_build/results?buildId=156184&view=results

**Pull request merged**